### PR TITLE
WIP: Demo add triggers

### PR DESCRIPTION
**Pull request by luftek**
In current situation triggers work for logic channels.
Would need to discuss some thing about soft-trigger module

pre-trigger capture ratio  pre-trigger capture ratio

### CODE REVIEW

File: src/hardware/demo/protocol.c [hunk 1]
```diff
@@ -456,6 +456,7 @@ SR_PRIV int demo_prepare_data(int fd, int revents, void *cb_data)
 	void *value;
 	uint64_t samples_todo, logic_done, analog_done, analog_sent, sending_now;
 	int64_t elapsed_us, limit_us, todo_us;
+	int trigger_offset, pre_trigger_samples;
```
Comment (abraxa) [inline src/hardware/demo/protocol.c]: Please use uint64_t or int64_t

File: src/soft-trigger.c [hunk 2]
```diff
@@ -35,7 +51,8 @@ SR_PRIV struct soft_trigger_logic *soft_trigger_logic_new(
 	stl = g_malloc0(sizeof(struct soft_trigger_logic));
 	stl->sdi = sdi;
 	stl->trigger = trigger;
-	stl->unitsize = (g_slist_length(sdi->channels) + 7) / 8;
+	/* Retreive number of logic channels, unitsize */
+	stl->unitsize = logic_channel_unitsize(sdi->channels);
```
Comment (abraxa) [inline src/soft-trigger.c]: What's wrong with the implementation that was there before? Why is logic_channel_unitsize() needed?

File: src/soft-trigger.c [hunk 3]
```diff
@@ -26,6 +26,22 @@
 #define LOG_PREFIX "soft-trigger"
 /* @endcond */
 
+static size_t logic_channel_unitsize(GSList *l)
+{
+	size_t number = 0;
+	struct sr_channel *channel;
+
+	for (channel = l->data; l; l = l->next)
+	{
```
Comment (abraxa) [inline src/soft-trigger.c]: This bracked belongs at the end of the previous line

File: src/hardware/demo/protocol.c [hunk 4]
```diff
@@ -515,19 +516,44 @@ SR_PRIV int demo_prepare_data(int fd, int revents, void *cb_data)
...
+			}
+		} else {
+			if (logic_done < samples_todo) {
+				sending_now = MIN(samples_todo - logic_done,
+						LOGIC_BUFSIZE / devc->logic_unitsize);
+				logic_generator(sdi, sending_now * devc->logic_unitsize);
+				trigger_offset = soft_trigger_logic_check(devc->stl,
+					devc->logic_data, sending_now * devc->logic_unitsize, &pre_trigger_samples);
+				
+				if (trigger_offset > -1) {
```
Comment (abraxa) [inline src/hardware/demo/protocol.c]: Brackets not needed here

File: src/soft-trigger.c (hunk 2 above)
Comment (luftek) [reply src/soft-trigger.c]: sdi->channels contains analog and logic channels. number of channels and unitsize can than be wrong.

**Comment by luftek** [comment]
I have incorporated your suggestions and need new review. Thank you.

**Comment by luftek** [comment]
Commited with abraxa rework

**Pull request closed without merging**